control/test_stand/generate_test_sequence.py

```diff
@@ -0,0 +1,153 @@
+import numpy as np
+from functools import partial
+import os
+
+
+MAX_ANGLE_RAD = 0.2268
+MS_PER_TICK = 25
+LENGTH_TICKS = 1000
+ALL_INDICES = {0, 1, 2, 3}
+
+
+def generate_yaml(command_list):
+    assert command_list.ndim == 2
+    assert command_list.shape[1] == 4
+    text = "commands:\n"
+    for cmd in command_list:
+        text += "  - [{servo_0}, {servo_1}, {servo_2}, {servo_3}]\n".format(
+            servo_0=cmd[0],
+            servo_1=cmd[1],
+            servo_2=cmd[2],
+            servo_3=cmd[3]
+        )
+
+    return text
+
+
+def write_file(text, path):
+    with open(path, 'w') as f:
+        f.write(text)
+
+
+def set_zero():
+    cmds = np.array([
+        [0.0, 0.0, 0.0, 0.0],
+        [0.0, 0.0, 0.0, 0.0],
+        [0.0, 0.0, 0.0, 0.0],
+    ])
+
+    return cmds
+
+
+def sine_wave_all(freq_hz):
+    milliseconds = np.arange(0.0, LENGTH_TICKS)
+
+    seconds = milliseconds * 1e-3
+
+    angular_freq_radps = freq_hz / (2.0 * np.pi)
+
+    return MAX_ANGLE_RAD * np.vstack([
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds)
+    ]).transpose()
+
+
+def symmetric(included, freq_hz=3.0):
+    milliseconds = np.arange(0.0, LENGTH_TICKS)
+
+    seconds = milliseconds * 1e-3
+
+    angular_freq_radps = freq_hz / (2.0 * np.pi)
+
+    result = MAX_ANGLE_RAD * np.vstack([
+        np.sin(angular_freq_radps * seconds),
+        np.sin(angular_freq_radps * seconds),
+        -np.sin(angular_freq_radps * seconds),
+        -np.sin(angular_freq_radps * seconds)
+    ]).transpose()
+
+    not_incls = ALL_INDICES - set(included)
+    for not_incl in not_incls:
+        result[:, not_incl] == 0.0
+
+    return result
+
+
+def antisymmetric(included, freq_hz=3.0):
```

Comment:
This appears the same as symmetric(...)?

Reply:
Recommend consolidating these methods.  Could multiply in a vector [1,1,1,1] or [1,1,-1,-1].  This would be more general.